test/service/internal_key_event_handlers/outdent_handler_test.dart

```diff
@@ -0,0 +1,268 @@
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import '../../infra/test_editor.dart';
+
+void main() async {
+  setUpAll(() {
+    TestWidgetsFlutterBinding.ensureInitialized();
+  });
+
+  group('outdent_handler.dart', () {
+    testWidgets(
+      "press shift tab in plain text",
+      (tester) async {
+        const text = 'Welcome to Appflowy 😁';
+        final editor = tester.editor
+          ..insertTextNode(text)
+          ..insertTextNode(text)
+          ..insertTextNode(text);
+
+        await editor.startTesting();
+
+        var document = editor.document;
+
+        var selection = Selection.single(path: [0], startOffset: 0);
+        await editor.updateSelection(selection);
+
+        await editor.pressLogicKey(
+          key: LogicalKeyboardKey.tab,
+          isShiftPressed: true,
+        );
+
+        // nothing happens
+        expect(
+          editor.documentSelection,
+          Selection.single(path: [0], startOffset: 0),
+        );
+        expect(editor.document.toJson(), document.toJson());
+
+        selection = Selection.single(path: [1], startOffset: 0);
+        await editor.updateSelection(selection);
+
+        await editor.pressLogicKey(
+          key: LogicalKeyboardKey.tab,
+          isShiftPressed: true,
+        );
+
+        // nothing happens
+        expect(
+          editor.documentSelection,
+          Selection.single(path: [1], startOffset: 0),
+        );
+        expect(editor.document.toJson(), document.toJson());
```

Comment:
```suggestion
        await editor.pressLogicKey(
          key: LogicalKeyboardKey.tab,
          isShiftPressed: true,
        );

        // nothing happens
        expect(
          editor.documentSelection,
          Selection.single(path: [1], startOffset: 0),
        );
        expect(editor.document.toJson(), document.toJson());
```
I'm not sure why you're testing the exact same behavior on the exact same type of textNode, you might as well assume that if it works on `path: [1]` then it would also work on `path: [0]` - Seeing as there is no way to go "further back" in `path: [0]`, it might be a redundant test-case.

Reply:
I just wrote it to be double sure, but we can remove it! Thanks again